### DISCUSSION TITLE
How can I format the code?

### DISCUSSION THREAD

**Discussion by IonutBalan123** [react-syntax-highlighter/react-syntax-highlighter#384] [Q&A]
Hello, want to build an app that shows code snippets and I'm wondering if it's possible to format the code, if a user posts the code inline, not formatted, and it will format itself, I searched on google but I didn't find a solution.

**Comment by mkrkelj1** [comment]
I use [Prettier](https://prettier.io/docs/en/api.html): 

```
npm install --save-dev --save-exact prettier
```

You will need to import the Babel parser as well:

```javascript
import prettier from 'prettier';
import parserBabel from 'prettier/parser-babel';

const codeString = "const Button = ({text}) => (<button>{text}</button>)"

const formatted = prettier.format(codeString, {
  semi: false,
  parser: "babel",
  plugins: [parserBabel],
});
```
